src/linker/utilities/slurm_utils.py

```diff
@@ -172,3 +121,102 @@ def submit_spark_cluster_job(
 
     session.deleteJobTemplate(jt)
     return master_error_log, jobs[0].split("_")[0]
+
+
+####################
+# Helper functions #
+####################
+
+
+def _get_cli_args(job_name, account, partition, peak_memory, max_runtime, num_threads):
+    return (
+        f"-J {job_name} "
+        f"-A {account} "
+        f"-p {partition} "
+        f"--mem={peak_memory*1024} "
+        f"-t {max_runtime}:00:00 "
+        f"-c {num_threads}"
+    )
+
+
+def _generate_job_template(
+    session: "drmaa.Session",
+    config: Config,
+    container_engine: str,
+    input_data: List[str],
+    results_dir: Path,
+    diagnostics_dir: Path,
+    step_id: str,
+    step_name: str,
+    implementation_name: str,
+    container_full_stem: str,
+    implementation_config: Optional[Dict[str, str]],
+) -> "drmaa.session.JobTemplate":
+    jt = session.createJobTemplate()
+    jt.jobName = f"{implementation_name}_{datetime.now().strftime('%Y%m%d%H%M%S')}"
+    jt.joinFiles = False  # keeps stdout separate from stderr
+    jt.outputPath = f":{str(diagnostics_dir / '%A.o%a')}"
+    jt.errorPath = f":{str(diagnostics_dir / '%A.e%a')}"
+    jt.remoteCommand = shutil.which("linker")
+    jt_args = [
+        "run-slurm-job",
+        container_engine,
+        str(results_dir),
+        str(diagnostics_dir),
+        step_id,
+        step_name,
+        implementation_name,
+        container_full_stem,
+        "-vvv",
+    ]
+    for filepath in input_data:
+        jt_args.extend(("--input-data", str(filepath)))
+    if implementation_config:
+        jt_args.extend(("--implementation-config", str(json.dumps(implementation_config))))
+    jt.args = jt_args
+    jt.jobEnvironment = {
+        "LC_ALL": "en_US.UTF-8",
+        "LANG": "en_US.UTF-8",
+    }
+    resources = config.slurm_resources
+    jt.nativeSpecification = _get_cli_args(
+        job_name=jt.jobName,
+        account=resources["account"],
+        partition=resources["partition"],
+        peak_memory=resources["memory"],
+        max_runtime=resources["time_limit"],
+        num_threads=resources["cpus"],
+    )
+
+    return jt
+
+
+def _generate_spark_cluster_job_template(
```

Comment:
I can  probably streamline this a bit w/ `_generate_job_template()` above if desired; I didn't feel compelled to since it'll all probably go away when we replace drmaa w/ snakemake anyway